Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -60,7 +77,57 @@ const ContestCategories = ({
         [ defaultSelected, categoriesFlat, getParents ],
     );
 
-    const handleTreeLabelClick = useCallback((node: ITreeItemType) => {
+    const strategyFilterGroup = useMemo(
+        () => possibleFilters.filter(({ type }) => type === FilterType.Strategy),
+        [ possibleFilters ],
+    );
+
+    const addNewStrategyFilters = useCallback(
+        (id: string, { allowedStrategyTypes }: ICategoryStrategiesTypes) => {
+            const strategyFiltersToAdd = allowedStrategyTypes
+                ?.map((value) => value.id.toString())
+                .map((x) => strategyFilterGroup.find((y) => y.value === x))
+                .filter((x) => !isNil(x)) as IFilter[];
```

Comment:
```
allowedStrategyTypes
                ?.map((value) => value.id.toString())
                .map((x) => strategyFilterGroup.find((y) => y.value === x))
                .filter((x) => !isNil(x)) as IFilter[];
```
Can be extracted into function with describable name
something like getStrategyFiltersToAdd